### ISSUE TITLE
Experiment with ejs template

### ISSUE THREAD

**Issue by designfrontier** [ansble/monument#297]
Using the config object changes and the render changes from #292 #291 run a site using ejs and see if there are things that need to be changed.

Also look at whether autodetection of the engine and if there are defaults that make sense if we can autodetect

**Comment by ExC0d3** [comment]
Hi,
I would like to try and work out this issue. Since I'm a first time contributor is there anything that I should keep in mind while handling this?
As of now this is my understanding of the problem - 

- Add a template object in the global config object to provide configuration for templating engine which by default is going to be dotJs (Eg. ejs/jade)
- Check if each render engine functions properly 

Sorry if this approach seems a bit over simplified.

**Comment by designfrontier** [comment]
Yeah that approach sounds pretty good. I have a branch up where I am working on getting the render items pulled into the config that you pass into the server. I am 90% of the way there (https://github.com/ansble/monument/pull/301) just have some failing tests to get squared away. But looking through that may give you some more insight.

I am hoping to find sometime this weekend to get that change finished up and merged into the next release branch.

You can probably start working with that branch as your base.

**Comment by designfrontier** [comment]
That branch being this one: https://github.com/ansble/monument/tree/feature/swappable-template-engines-add-config-objects

**Comment by designfrontier** [comment]
also the config object would include the template engine itself, so the ejs engine as well as any config that it needs.

**Comment by designfrontier** [comment]
If you start from the Strade Bianchi branch you should be able to make a
run at it now. Got the config object pull request merged over the weekend
